src/config.js

```diff
@@ -44,7 +45,11 @@ class Config {
     }
 
     if (this.input.mode === 'start') {
-      if (!this.input.ec2ImageId || !this.input.ec2InstanceType || !this.input.subnetId || !this.input.securityGroupId || !this.input.ec2BaseOs) {
+      const isSet = param => param;
+      const instanceParams = [this.input.ec2ImageId, this.input.ec2InstanceType, this.input.ec2BaseOs, this.input.subnetId, this.input.securityGroupId];
+      const templateParams = [this.input.ec2LaunchTemplate, this.input.ec2BaseOs];
+
+      if (!(instanceParams.every(isSet) || templateParams.every(isSet) )) {
```

Comment:
I'm not sure I understand your statement on why this is not working. Like in your example, if instanceParams.every() is false, templateParams.every() is true, that will results a false, because the overall result is negated. It is tricky condition, but the reason of this separation to support either launch template options or normal instance options. Launch template option is quite unique, because all the instanceParams that you specifiy additionally are acting as override value in launch template.

Reply:
I'm also not sure I understand your "0" comment, because I'm not sure how zero value can set. Inputs are all strings, because that's what the core.getInput() returns. Could you please clarify this more for me please?